Firestore/core/src/firebase/firestore/local/leveldb_remote_document_cache.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_LOCAL_LEVELDB_REMOTE_DOCUMENT_CACHE_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_LOCAL_LEVELDB_REMOTE_DOCUMENT_CACHE_H_
+
+#if !defined(__OBJC__)
+#error "For now, this file must only be included by ObjC source files."
+#endif  // !defined(__OBJC__)
+
+#import <Foundation/Foundation.h>
+
+#include <vector>
+
+#include "Firestore/core/src/firebase/firestore/model/document_key.h"
+#include "Firestore/core/src/firebase/firestore/model/document_key_set.h"
+#include "Firestore/core/src/firebase/firestore/model/document_map.h"
+#include "Firestore/core/src/firebase/firestore/model/types.h"
+#include "absl/strings/string_view.h"
+
+@class FSTLevelDB;
+@class FSTLocalSerializer;
+@class FSTMaybeDocument;
+@class FSTQuery;
+
+NS_ASSUME_NONNULL_BEGIN
+
+namespace firebase {
+namespace firestore {
+namespace local {
+
+/** Cached Remote Documents backed by leveldb. */
+class LevelDbRemoteDocumentCache {
+ public:
+  LevelDbRemoteDocumentCache(FSTLevelDB *db, FSTLocalSerializer *serializer);
+
+  void AddEntry(FSTMaybeDocument *document);
```

Comment:
Nit: in C++, we're following the opposite convention: `&` and `*` are attached to the type name, not the variable name (applies throughout). Unfortunately, AFAIK `clang-formatter` uses heuristics for this to find out which style is more prevalent in the file and use that, so this would have to be fixed manually.

Reply:
Got it. Fixed.